fuseimg/data/ops/shape_ops.py

```diff
@@ -82,7 +81,214 @@ def __call__(self, sample_dict: NDict, op_id: Optional[str], key: str,
+        All blocks above a threshold (10) are considered non-empty.
+        Then, the biggest connected component (at the blocks level) is extracted, and its axis-aligned bbox is returned.
+        :param img: Image instance , expected 2d breast integer grayscale image where 0 is black background color
+        :param dark_area_threshold: defines grayscale level from which lower is consider dark / outside of body
+        :param blocks_num: number of blocks the algorithm split the images into
+        :return: four coordinates
+        """
+        bl_rows = img.shape[0]//self.blocks_num
+        bl_cols = img.shape[1]//self.blocks_num
+
+        rows_starts = list(range(self.blocks_num))
+        cols_starts = list(range(self.blocks_num))
+
+        rows_starts = list(map(lambda x:x*bl_rows, rows_starts))
+        cols_starts = list(map(lambda x: x * bl_cols, cols_starts))
+
+        cells = np.zeros((self.blocks_num,self.blocks_num))
+
+        for ri,r in enumerate(rows_starts):
+            for ci,c in enumerate(cols_starts):
+                r_end = min(img.shape[0]-1,r+bl_rows)
+                c_end = min(img.shape[1] - 1, c + bl_cols)
+                cells[ri,ci] = np.mean(img[r:r_end,c:c_end])
+
+        cells_binary = np.zeros((self.blocks_num,self.blocks_num),dtype=bool)
+        cells_binary[cells>self.dark_area_threshold] = True #TODO: this is a hardcoded threshold, see if there's a better alternative
+        blobs_labels = measure.label(cells_binary, background=0)
+        regions = measure.regionprops(blobs_labels)
+
+        regions_areas = [r.area for r in regions]
+        if len(regions)<1:
+            # expected to have at least background and one object. TODO: consider reporting error and returning AABB of the full image.
+            print('Warning: could not crop properly! fallbacking to full image')
+            return 0,0,img.shape[0]-1,img.shape[1]-1
+
+
+        for i,r in enumerate(regions):
+            if r.label==0:
+                regions_areas[i]=-1
+                break
+        max_ind = np.argmax(regions_areas)
+        bbox = regions[max_ind].bbox
+
+        full_img_bbox = [bbox[0]*bl_rows,bbox[1]*bl_cols,(bbox[2]+1)*bl_rows,(bbox[3]+1)*bl_cols]
+        minr, minc, maxr, maxc = full_img_bbox
+        maxr = min(maxr, img.shape[0]-1)
+        maxc = min(maxc, img.shape[1]-1)
+
+        return minr, minc, maxr, maxc
+
+    def __call__(self, sample_dict: NDict, key: str
+        ):
+        '''
+        ''' 
         
+        img = sample_dict[key]
+        aabb = self.find_breast_aabb(img)
+        img = img[aabb[0]: aabb[2], aabb[1]: aabb[3]].copy()
+        sample_dict[key] = img
+        return sample_dict
+    
+class OpFlipBrightSideOnLeft2D(OpBase):
+    '''
+     checks if the image needed to be flipped to the left
+    '''
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.max_pixel_value = 255.0
```

Comment:
Do you expect the image to be between 0:255? Do expect a gray scale?.
Specify all your assumptions in the comments please.

Reply:
grayscale, it was already in the parameter documentation